### PR TITLE
Add vcpupin support

### PR DESCRIPTION
**Pull request by mark-i-m**

### CODE REVIEW

**Comment by mark-i-m** [comment]
Locally, when I attempt to test, I get the following error:

```console
$ bundle exec vagrant up --provider=libvirt
You appear to be running Vagrant outside of the official installers.
Note that the installers are what ensure that Vagrant has all required
dependencies, and Vagrant assumes that these dependencies exist. By
running outside of the installer environment, Vagrant may not function
properly. To remove this warning, install Vagrant using one of the
official packages from vagrantup.com.

Bringing machine 'test_vm_test' up with 'libvirt' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

Libvirt Provider:
* The following settings shouldn't exist: cpuaffinity
```

File: README.md [hunk 1]
```diff
@@ -290,6 +290,20 @@ end
   end
   ```
 
+* `cpuaffinitiy` - Mapping of vCPUs to host CPUs. [See `vcpupin`](https://libvirt.org/formatdomain.html#elementsCPUTuning).
+
+  ```ruby
+  Vagrant.configure("2") do |config|
+    config.vm.provider :libvirt do |libvirt|
+      libvirt.cpus = 4
+      libvirt.cpuaffinitiy
```
Comment (electrofelix) [inline README.md:299]: Typo using `cpuaffinity` here, but `cpu_affinity` for the config option below. Guessing that is the source of the failure you were seeing testing locally?

**Review by electrofelix — COMMENTED** [review]
catching up on some older PRs, spotted a bug in how the config option was named versus referenced.

**Comment by acidghost** [comment]
@mark-i-m if you're working on this PR anymore I'd be interested to finish it up

**Comment by mark-i-m** [comment]
@acidghost Please do! I unfortunately no longer have the time :/

@electrofelix Thanks for creating this project. It's a useful tool. I wish I had time to work on it.

**Pull request merged**